### ISSUE TITLE
Add rule to enforce returning calls that return 'never'

### ISSUE THREAD

**Issue by ajafff** [fimbullinter/wotan#154]
```ts
declare assertNever(v: never): never;

if (foo) {
  // do stuff
} else {
  assertNever(foo); // forgot to return here
  return foo; // dead code not detected
}
```
